src/analytics/src/main/scala/osmesa/analytics/oneoffs/ChangesetORCCreator.scala

```diff
@@ -0,0 +1,181 @@
+
+/*
+ * Usage example:
+ *
+ * sbt "project analytics" assembly
+ *
+ * spark-submit \
+ *   --class osmesa.analytics.oneoffs.ChangesetORCUpdater \
+ *   ingest/target/scala-2.11/osmesa-analytics.jar \
+ *   --changeset-source http://somewhere/diffs/ \
+ *   --database-url $DATABASE_URL
+ */
+object ChangesetORCUpdater
+  extends CommandApp(
+    name = "osmesa-changeset-orc-updater",
+    header = "Bring existing changesets ORC file up to date using changeset stream",
+    main = {
+
+      import ChangesetORCUpdaterUtils._
+
+      val changesetSourceOpt =
+        Opts
+          .option[URI](
+          "changesets",
+          short = "c",
+          metavar = "uri",
+          help = "Location of replication changesets"
+        )
+        .validate("Changeset source must have trailing '/'") { _.getPath.endsWith("/") }
+
+      val endTimeOpt =
+        Opts
+          .option[Instant]("end-time",
+                       short = "e",
+                       metavar = "timestamp",
+                       help = "Timestamp of stream end (of the form 2016-02-29T13:45:00Z); if absent, the time now will be used")
+                         .orNone
+
+      val orcArg = Opts
+        .argument[URI]("source ORC")
+        .validate("URI to ORC must have an s3 or file scheme") { _.getScheme != null }
+        .validate("orc must be an S3 or file Uri") { uri =>
+          uri.getScheme.startsWith("s3") || uri.getScheme.startsWith("file")
+        }
+        .validate("orc must be an .orc file") { _.getPath.endsWith(".orc") }
+
+      val outputArg = Opts.argument[URI]("destination ORC")
+        .validate("Output URI must have a scheme") { _.getScheme != null }
+        .validate("Output URI must have an S3 or file scheme") { uri =>
+          uri.getScheme.startsWith("s3") || uri.getScheme.startsWith("file")
+        }
+        .validate("orc must be an .orc file") { _.getPath.endsWith(".orc") }
+
+      (changesetSourceOpt,
+       endTimeOpt,
+       orcArg,
+       outputArg).mapN {
+        (changesetSource, endTime, orcUri, outputURI) =>
+        implicit val spark: SparkSession = Analytics.sparkSession("ChangesetStatsUpdater")
+
+        import spark.implicits._
+
+        val df = spark.read.orc(orcUri.toString)
+        val lastModified = df.select(max(coalesce('closed_at, 'created_at))).first.getAs[Timestamp](0)
+
+        val startSequence = findSequenceFor(lastModified.toInstant, changesetSource)
+        val endSequence = endTime.map(findSequenceFor(_, changesetSource)).getOrElse(getCurrentSequence(changesetSource).sequence)
+
+        val options = Map(
+          Source.BaseURI -> changesetSource.toString,
+          Source.StartSequence -> startSequence.toString,
+          Source.EndSequence -> (endSequence + 1).toString // sequence range is (]; end sequence is exclusive
+        )
+
+        val changesets = spark.read.format(Source.Changesets).options(options).load
+        changesets
+          .drop("comments", "sequence")
```

Comment:
I wonder if it's better to drop these columns, or to somehow preserve them, though, they aren't currently in the table schema...

Reply:
I would love to preserve them in the combined output. We don't do anything with comments (and they're not in the dump we typically convert), but I can see eventual value. For `sequence`, we can let it be `null` for rows that came from a dump.

In theory, the probably `sequence` could be calculated based on the timestamp, but probably best to leave that to downstream consumers.

Reply:
This item isn't demanding action at present, so I'll ignore it for the purposes of this PR.  But if there's something specific to do in relation to this point, make an issue in VP?